resources/js/Shared/Components/DeleteModal.vue

```diff
@@ -0,0 +1,109 @@
+<script setup>
+import { ref, computed } from 'vue'
+import { Dialog, DialogPanel, DialogTitle, TransitionChild, TransitionRoot } from '@headlessui/vue'
+import { ExclamationTriangleIcon, XMarkIcon } from '@heroicons/vue/24/outline'
+import { __ } from '@/translate'
+
+const emits = defineEmits(['close', 'delete'])
+
+const closeModal = () => emits('close')
+const confirmDelete = () => emits('delete')
+const open = ref(true)
+
+const props = defineProps({
+  name: {
+    type: String,
+    required: true,
+  },
+  type: {
+    type: String,
+    required: true,
+  },
+})
+
+const formattedType = computed(() => {
+  return props.type.charAt(0).toLowerCase() + props.type.slice(1)
+})
+
+const renderHeader = () => {
+  let translationKey = ''
+  translationKey = 'Delete ' + formattedType.value + '?'
+
+  return __(translationKey, { type: props.type })
+}
+
+const renderText = () => {
+  let translationKey = ''
+
+  if (props.type === 'City') {
+    translationKey = 'City :name will be permanently deleted.'
+  } else {
+    translationKey = 'Country :name and its cities will be permanently deleted.'
+  }
```

Comment:
What if you will have 100 resources to delete? Will be  there 100 if else statements?
It should be refactored.
Maybe pass the whole message prop to the modal?

Reply:
I removed if else statement